Bottom: 19c2f3b968a25bc73a607563b2b20d761eaee505
Top:    5ec1a23b32d8c72dbd906a244c299ca6a3c47c9b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-28 03:11:24 +0100

Refresh of use-trackball-to-allow-rotatio

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index df9383b..9ba6009 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -246,9 +246,12 @@ ghid_note_event_location (GdkEventButton * ev)
       event_y = ev->y;
     }
 
+  /* FIXME: IFDEF HACK */
+#ifdef ENABLE_GL
   /* Unproject event_x and event_y to world coordinates of the plane we are on */
   ghid_unproject_to_z_plane (event_x, event_y, 0,
                              &event_x, &event_y);
+#endif
 
   gport->view_x = event_x * gport->zoom + gport->view_x0;
   gport->view_y = event_y * gport->zoom + gport->view_y0;
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index d0aae74..faf7406 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -2153,7 +2153,10 @@ ghid_build_pcb_top_window (void)
   GtkWidget *vbox_main, *vbox_left, *hbox_middle, *hbox = NULL;
   GtkWidget *viewport, *ebox, *vbox, *frame;
   GtkWidget *label;
+  /* FIXME: IFDEF HACK */
+#ifdef ENABLE_GL
   GtkWidget *trackball;
+#endif
   GHidPort *port = &ghid_port;
   gchar *s;
   GtkWidget *scrolled;
@@ -2262,12 +2265,15 @@ ghid_build_pcb_top_window (void)
       GTK_POLICY_NEVER, GTK_POLICY_AUTOMATIC);
   make_layer_buttons(vbox, port);
 
+  /* FIXME: IFDEF HACK */
+#ifdef ENABLE_GL
   trackball = ghid_trackball_new ();
   g_signal_connect (trackball, "rotation-changed",
                     G_CALLBACK (ghid_port_rotate), NULL);
   g_signal_connect (trackball, "view-2d-changed",
                     G_CALLBACK (ghid_view_2d), NULL);
   gtk_box_pack_start (GTK_BOX (vbox_left), trackball, FALSE, FALSE, 0);
+#endif
 
   vbox = gtk_vbox_new(FALSE, 0);
   gtk_box_pack_start(GTK_BOX(vbox_left), vbox, FALSE, FALSE, 0);
